hera_sim/visibilities/vis_cpu.py

```diff
@@ -34,8 +33,20 @@ def __init__(self, bm_pix=100, real_dtype=np.float32,
             Arguments of :class:`VisibilitySimulator`.
         """
 
-        self._real_dtype = real_dtype
-        self._complex_dtype = complex_dtype
+        assert precision in (1,2)
+        self._precision = precision
+        if precision == 1:
+            self._real_dtype = np.float32
+            self._complex_dtype = np.complex64
+        else:
+            self._real_dtype = np.float64
+            self._complex_dtype = np.complex128
```

Comment:
Are these definitions used at all? Since you've changed the signature of `vis_cpu` to also take the `precision` parameter (which I think is much cleaner!), you don't need these dtypes at all.

Reply:
Thw class builds arrays that get passed to vis_cpu/vis_gpu, and it's more efficient for them to be built in the correct datatype, so I think these should stay.